tests/test_rss.py

```diff
@@ -0,0 +1,266 @@
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.core.rss import (RSS,
+                                            RSSCommand,
+                                            RSSClient)
+
+
+RSS_FEED_URL = 'http://example.com/rss'
+
+requests_http = []
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+class TestRSSBackend(unittest.TestCase):
+    """RSS backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        rss = RSS(RSS_FEED_URL, tag='test')
+
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, 'test')
+        self.assertIsInstance(rss.client, RSSClient)
+
+        # When tag is empty or None it will be set to
+        # the value in url
+        rss = RSS(RSS_FEED_URL)
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, RSS_FEED_URL)
+
+        rss = RSS(RSS_FEED_URL, tag='')
+        self.assertEqual(rss.url, RSS_FEED_URL)
+        self.assertEqual(rss.origin, RSS_FEED_URL)
+        self.assertEqual(rss.tag, RSS_FEED_URL)
+
+    def test_has_caching(self):
+        """Test if it returns True when has_caching is called"""
+
+        self.assertEqual(RSS.has_caching(), True)
+
+    def test_has_resuming(self):
+        """Test if it returns False when has_resuming is called"""
+
+        self.assertEqual(RSS.has_resuming(), False)
+
+    def __configure_http_server(self):
+        bodies_entries_job = read_file('data/rss_entries.xml')
+
+        def request_callback(method, uri, headers):
+            if uri.startswith(RSS_FEED_URL):
+                body = bodies_entries_job
+            else:
+                body = ''
+
+            requests_http.append(httpretty.last_request())
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               RSS_FEED_URL,
+                               responses=[
+                                    httpretty.Response(body=request_callback) \
+                                    for _ in range(2)
+                               ])
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether a list of entries is returned"""
+
+        self.__configure_http_server()
+
+        # Test fetch entries from feed
+        rss = RSS(RSS_FEED_URL)
+        entries = [item for item in rss.fetch()]
+        self.assertEqual(len(entries), 30)
+
+        # Test metadata
+        expected = [('98572defb3a652afbfdfe96517edefb88a22dcfa', 1481044620.0),
+                    ('e3b0d7463fca0c47d82debc3ddc37d8906f77548', 1480955040.0),
+                    ('28bca39353ff0825f53b157b69da319e2f49ab4d', 1480886040.0)]
+
+        for x in range(len(expected)):
```

Comment:
You should check some of the elements of the parsed data (such as the title), not just the metadata fields.